apps/core/src/lifecycle-hooks/dynamodb-local-setup.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  CreateTableCommand,
+  CreateTableCommandInput,
+  DynamoDBClient,
+  ResourceInUseException,
+} from '@aws-sdk/client-dynamodb';
+import {
+  Inject,
+  Injectable,
+  Logger,
+  OnApplicationBootstrap,
+} from '@nestjs/common';
+import { ConfigType } from '@nestjs/config';
+import { launch as ddbLocalLaunch } from 'dynamodb-local';
+import { databaseConfig } from '../config/database.config';
+
+const apiResourceTable =
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  require('../../api-resource-table-properties.js') as CreateTableCommandInput;
+const TABLE_CREATION_ATTEMPT_NUMBER = 30;
+const TABLE_CREATION_ATTEMPT_DELAY = 1000; // in milliseconds
+
+@Injectable()
+export class DynamoDbLocalSetupService implements OnApplicationBootstrap {
+  private readonly logger = new Logger(DynamoDbLocalSetupService.name);
+
+  constructor(
+    @Inject(databaseConfig.KEY)
+    private dbConfig: ConfigType<typeof databaseConfig>,
+  ) {}
+
+  async onApplicationBootstrap() {
+    this.logger.log('Launching DynamoDB local instance');
+    await ddbLocalLaunch(this.dbConfig.port);
+    this.logger.log('DynamoDB local instance launched');
+    await this.createApiResourceTable();
+  }
+
+  private async createApiResourceTable() {
+    const ddbClient = new DynamoDBClient({ endpoint: this.dbConfig.endpoint });
+    let attempt = 0;
+
+    do {
```

Comment:
Why do we need to retry table creation?

Reply:
It is waiting for the local instance to spin up. The logic could be separated into [1] wait for spin up, and then [2] table creation request when we have more requests.